gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/PlanServiceImpl.java

```diff
@@ -270,6 +289,10 @@ public PlanEntity update(final ExecutionContext executionContext, UpdatePlanEnti
 
             newPlan.setCharacteristics(updatePlan.getCharacteristics());
 
+            final var apiId = newPlan.getApi();
+            Api api = apiRepository.findById(apiId).orElseThrow(() -> new ApiNotFoundException(apiId));
```

Comment:
do you think caller can pass the Api object directly instead of doing a findById ?